crd/src/lib.rs

```diff
@@ -187,7 +187,8 @@ pub struct ConfigOption {
 
 #[derive(Clone, Debug, Deserialize, Eq, JsonSchema, PartialEq, Serialize)]
 pub struct SparkClusterStatus {
-    pub version: SparkVersion,
+    pub current_version: SparkVersion,
+    pub target_version: SparkVersion,
```

Comment:
Yeah it makes sense. I think its still close to my approach (at least in terms of logic). We keep the target version to avoid updating while updating etc. I will change to Option. It is more logical to have no target version set if no updates are required. With my approach, current and target version just dangle with the same value in the status while no updates are required.